kilkari/src/main/java/org/motechproject/nms/kilkari/domain/InboxCallDetailRecord.java

```diff
@@ -47,11 +48,14 @@
     @Field
     private Set<InboxCallData> content;
 
-    public InboxCallDetailRecord() { }
+    public InboxCallDetailRecord() {
+        content = new HashSet<>();
+    }
 
     public InboxCallDetailRecord(Long callingNumber, String operator, String circle, // NO CHECKSTYLE More than 7 parameters
                                  Long callId, DateTime callStartTime, DateTime callEndTime, Integer callDurationInPulses,
                                  Integer callStatus, Integer callDisconnectReason, Set<InboxCallData> content) {
+        this();
```

Comment:
I'm with you on the base constructor for inheritance and the this(...) in the sense of calling a parameterized constructor from a no param constructor - like this(a, 10) from no param.

but this seems a little backwards to me in the sense of calling a no param constructor from a parameterized one and splitting up the initialization between 2 constructors.

Not a deal breaker though.